src/boot/efi/boot.c

```diff
@@ -845,13 +875,23 @@ static BOOLEAN menu_run(
 
         *chosen_entry = config->entries[idx_highlight];
 
+        /* The user is likely to cycle through several modes before
+         * deciding to keep one. Therefore, we update the EFI var after
+         * we left the menu to reduce nvram writes. */
+        if (update_console_mode_efivar) {
+                if (config->console_mode_efivar == CONSOLE_MODE_KEEP)
+                        efivar_set(LOADER_GUID, L"LoaderConfigConsoleMode", NULL, EFI_VARIABLE_NON_VOLATILE);
+                else
+                        efivar_set_uint_string(LOADER_GUID, L"LoaderConfigConsoleMode",
```

Comment:
maybe suppress this cll if what we loaded from the efivar originally matches what we are about to set? i.e. for cases where people cycled through all modes, figured out "fuck, default was good", and thus this shouldn't result in an nvram write.